[CI] LogsdbRestIT testSyntheticSourceRuntimeFieldQueries failing

**Build Scans:**
- [elasticsearch-intake #41607 / part2](https://gradle-enterprise.elastic.co/s/fcvv45iafunsk)
- [elasticsearch-pull-request #154608 / part-2](https://gradle-enterprise.elastic.co/s/svqe2f2s3euti)
- [elasticsearch-pull-request #154607 / part-2](https://gradle-enterprise.elastic.co/s/nizmx2cwsgtrw)
- [elasticsearch-pull-request #154611 / part-2](https://gradle-enterprise.elastic.co/s/fejif24lcqsza)
- [elasticsearch-pull-request #154608 / part-2](https://gradle-enterprise.elastic.co/s/ycj2imylocabi)
- [elasticsearch-intake #41606 / part2](https://gradle-enterprise.elastic.co/s/xymina4d2553y)
- [elasticsearch-pull-request #154601 / part-2](https://gradle-enterprise.elastic.co/s/rs2a2guayioey)
- [elasticsearch-intake #41604 / part2](https://gradle-enterprise.elastic.co/s/77yylky7uncvm)
- [elasticsearch-pull-request #154581 / part-2](https://gradle-enterprise.elastic.co/s/qvmsruorsi7mi)
- [elasticsearch-pull-request #154592 / part-2](https://gradle-enterprise.elastic.co/s/wnmfeli363njo)

**Reproduction Line:**
```
./gradlew ":x-pack:plugin:logsdb:javaRestTest" --tests "org.elasticsearch.xpack.logsdb.LogsdbRestIT.testSyntheticSourceRuntimeFieldQueries" -Dtests.seed=EFB795870C8B3229 -Dtests.locale=lmo-Latn-IT -Dtests.timezone=US/East-Indiana -Druntime.java=26
```

**Applicable branches:**
main

**Reproduces locally?:**
N/A

**Failure History:**
[See dashboard](https://es-delivery-stats.elastic.dev/app/dashboards#/view/dcec9e60-72ac-11ee-8f39-55975ded9e63?_g=(refreshInterval:(pause:!t,value:60000),time:(from:now-7d%2Fd,to:now))&_a=(controlGroupState:(initialChildControlState:('0c0c9cb8-ccd2-45c6-9b13-96bac4abc542':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:task.keyword,order:0,selectedOptions:!(),title:'GradleTask',type:optionsListControl),'4e6ad9d6-6fdc-4fcc-bf1a-aa6ca79e0850':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:className.keyword,order:1,selectedOptions:!(org.elasticsearch.xpack.logsdb.LogsdbRestIT),title:'Suite',type:optionsListControl),'144933da-5c1b-4257-a969-7f43455a7901':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:name.keyword,order:2,selectedOptions:!('testSyntheticSourceRuntimeFieldQueries'),title:'Test',type:optionsListControl)))))

**Failure Message:**
```
java.lang.AssertionError: 
Expected: map containing ["errors"-><false>]
     but: map was [<took=0>, <items=[{create={_index=logs-test-esql-synthetic-foo, _id=null, error={reason=mapping-level runtime fields are not allowed in index using [logsdb_columnar] index mode, type=illegal_argument_exception}, status=400}}, {create={_index=logs-test-esql-synthetic-foo, _id=null, error={reason=mapping-level runtime fields are not allowed in index using [logsdb_columnar] index mode, type=illegal_argument_exception}, status=400}}, {create={_index=logs-test-esql-synthetic-foo, _id=null, error={reason=mapping-level runtime fields are not allowed in index using [logsdb_columnar] index mode, type=illegal_argument_exception}, status=400}}, {create={_index=logs-test-esql-synthetic-foo, _id=null, error={reason=mapping-level runtime fields are not allowed in index using [logsdb_columnar] index mode, type=illegal_argument_exception}, status=400}}, {create={_index=logs-test-esql-synthetic-foo, _id=null, err
[truncated]
```

**Issue Reasons:**
- [main] 4 consecutive failures in test testSyntheticSourceRuntimeFieldQueries
- [main] 3 consecutive failures in step part2
- [main] 3 consecutive failures in step part-2
- [main] 1 consecutive failures in step checkpart2_gpu-tests
- [main] 3 consecutive failures in pipeline elasticsearch-intake
- [main] 3 consecutive failures in pipeline elasticsearch-pull-request
- [main] 25 failures in test testSyntheticSourceRuntimeFieldQueries (7.4% fail rate in 336 executions)
- [main] 4 failures in step part2 (7.5% fail rate in 53 executions)
- [main] 20 failures in step part-2 (8.6% fail rate in 232 executions)
- [main] 4 failures in pipeline elasticsearch-intake (7.5% fail rate in 53 executions)
- [main] 12 failures in pipeline elasticsearch-pull-request (5.3% fail rate in 226 executions)

**Note:**
This issue was created using new test triage automation. Please report issues or feedback to es-delivery.

--- a/muted-tests.yml
+++ b/muted-tests.yml
@@ -427,12 +427,6 @@ tests:
 - class: org.elasticsearch.xpack.esql.qa.multi_node.GenerativeIT
   method: test {feature:SUBQUERIES}
   issue: https://github.com/elastic/elasticsearch/issues/149681
-- class: org.elasticsearch.xpack.logsdb.LogsdbRestIT
-  method: testSyntheticSourceRuntimeFieldQueries
-  issue: https://github.com/elastic/elasticsearch/issues/151546
-- class: org.elasticsearch.xpack.logsdb.LogsdbRestIT
-  method: testEsqlRuntimeFields
-  issue: https://github.com/elastic/elasticsearch/issues/151547
 - class: org.elasticsearch.xpack.esql.qa.multi_node.GenerativeIT
   method: test {feature:BASELINE}
   issue: https://github.com/elastic/elasticsearch/issues/149746

